Detections/MultipleDataSources/PhishinglinkExecutionObserved.yaml

```diff
@@ -0,0 +1,97 @@
+severity: Medium
+requiredDataConnectors:
+  - connectorId:  OfficeATP
+    dataTypes:
+    - SecurityAlert
+  - connectorId:  PaloAltoNetworks
+    dataTypes:
+    - CommonSecurityLog (PaloAlto)
+  - connectorId:  Fortinet
+    dataTypes:
+    - CommonSecurityLog (Fortinet)
+  - connectorId:  CheckPoint
+    dataTypes:
+    - CommonSecurityLog (CheckPoint)
+  - connectorId:  Zscaler
+    dataTypes:
+    - CommonSecurityLog (Zscaler)
+queryFrequency: 1d
+queryPeriod: 1d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - InitialAccess
+relevantTechniques:
+  - T1566
+query: |
+    //Finding MDO Security alerts and extracting the Entities user, Domain, Ip, and URL. 
+    SecurityAlert
+    |where ProviderName =~ "Office 365 Advanced Threat Protection"
+    //extracting Alert Entities 
+    | extend Entities = parse_json(Entities)
+    | mv-apply Entity = Entities on
+    (
+    where Entity.Type == 'account'
+    | extend EntityUPN = iff(isempty(Entity.UserPrincipalName), tostring(strcat(Entity.Name, "@", tostring (Entity.UPNSuffix))), tostring(Entity.UserPrincipalName))
+    )
+    | mv-apply Entity = Entities on
+    (
+    where Entity.Type == 'ip'|extend EntityIp = tostring(Entity.Address)
+    )
+    | mv-apply Entity = Entities on
+    (
+    where Entity.Type == 'host'| extend EntityHost = tostring(Entity.HostName)
+    )
+    | mv-apply Entity = Entities on
+    (
+    where Entity.Type == 'url'
+    | extend EntityUrl = tostring(Entity.Url)
+    )
+    | where not(ipv4_is_private(EntityIp))
+    | summarize
+    AccountUpn=take_any(EntityUPN),IPAddress=take_any(EntityIp),HostName=take_any(EntityHost),Url=tostring(tolower(take_any(EntityUrl))),AlertTime= min(TimeGenerated)by SystemAlertId, ProductName
+    // filtering 3pnetwork devices 
+    | join kind= inner (CommonSecurityLog
+    | where DeviceVendor has_any  ("Palo Alto Networks", "Fortinet", "Check Point", "Zscaler")
+    | where DeviceAction != "Block"
+    | where DeviceProduct startswith "FortiGate" or DeviceProduct startswith  "PAN" or DeviceProduct startswith  "VPN" or DeviceProduct startswith "FireWall" or DeviceProduct startswith  "NSSWeblog" or DeviceProduct startswith "URL"
+    | where isnotempty(DestinationHostName) 
+    | where isnotempty(RequestURL)
+    | project
+    3plogTime=TimeGenerated,
+    DeviceVendor,
+    DeviceProduct,
+    Activity,
+    DestinationHostName,
+    DestinationIP,
+    RequestURL=tostring(tolower(RequestURL)),
+    MaliciousIP,
+    Name = tostring(split(SourceUserName,"@")[0]),
+    UPNSuffix =tostring(split(SourceUserName,"@")[1]),
+    SourceUserName,
+    IndicatorThreatType,
+    ThreatSeverity,AdditionalExtensions,
+    ThreatConfidence)on $left.Url == $right.RequestURL
```

Comment:
Do we also want to try and match on user name to see if phished user == the user who clicked?

Reply:
Changes added